src/main/java/pl/gov/coi/pomocua/ads/materialaid/MaterialAidResource.java

```diff
@@ -26,9 +26,11 @@ public MaterialAidOffer postMaterialAidOffer(
         return repository.save(materialAidOffer);
     }
 
+    @Operation(description = "Allows to search for material aid offers using different criteria (passes as query params). Each criteria is optional.")
     @GetMapping("material-aid")
-    public Page<MaterialAidOffer> list(Pageable pageRequest) {
-        return repository.findAll(pageRequest);
+    public Page<MaterialAidOffer> list(Pageable pageRequest, MaterialAidOfferSearchCriteria searchCriteria) {
+        Page<MaterialAidOffer> all = repository.findAll(MaterialAidOfferSpecifications.from(searchCriteria), pageRequest);
```

Comment:
Unnecessary local var -> just `return repo.findAll(...`

Reply:
I forgot to revert changes after debugging :-) thanks

Reply:
done